### ISSUE TITLE
Consider deprecating and removing IO/mirrors libraries

### ISSUE THREAD

**Issue by cbracken** [google/quiver-dart#621]
The IO library contains three functions. I did a quick scan of internal usage at Google and found that across hundreds of thousands of imports of quiver, these have almost no usage whatsoever; probably because most of our code is either web or Flutter.

Eliminating this has the nice effect of making Quiver show as a web-friendly package on pub.dev. If we also eliminated or split out the mirrors package it would show as flutter-compatible too. If we think they're worth keeping perhaps we could split them out into a separate package, or find them a home in an existing package.

## byteStreamToString
```dart
//  Converts a [Stream] of byte lists to a [String].
Future<String> byteStreamToString(Stream<List<int>> stream, {Encoding encoding = utf8}) {
  return stream.transform(encoding.decoder).join();
}
```
I found one usage in Fuchsia.

This doesn't actually depend on dart:io, but I suspect we hadn't yet created the async library when this was landed. If we want to keep this, we can move it to the async library, forward it here and deprecate this one until we delete it.

##  getFullPath
```dart
String getFullPath(path) => File(path).resolveSymbolicLinksSync();
```
I found no uses of this internally.

This is potentially useful in a case like `listOfPaths.map(getFullPath)` but it's also a one-liner that existing users could easily inline into their code if they like it.

## visitDirectory

```dart
/// Lists the sub-directories and files of this Directory, optionally recursing
/// into sub-directories based on the return value of [visit].
///
/// [visit] is called with a [File], [Directory] or [Link] to a directory,
/// never a Symlink to a File. If [visit] returns true, then its argument is
/// listed recursively.
Future visitDirectory(Directory dir, Future<bool> visit(FileSystemEntity f)) {
  ...
}
```
I found no uses of this internally.

This is potentially useful, but the fact that we don't use it, even within our Dart tooling, suggests that it's probably not worth keeping around. If there's another package where it might be a good fit we could contribute it there.

/cc @yjbanov @jtmcdole @adambender @a14n for thoughts.

**Comment by jtmcdole** [comment]
I've written a "visitDirectory" I think about twice in the past, and I'd
probably just copy-paste that anytime I needed it. If we had a
package:filesystem, it would go great there... not generally a quivery
thing though. Also, the "visitor" pattern is well known.

`collectStream<T>` could just go in async. The number of times I've wanted
to go from Stream<List<T>> to List<T>....

**Comment by cbracken** [comment]
Closed by #628 and #629.

**Comment by cbracken** [comment]
Deleted from the 3.0 branch in #631.
